.github/workflows/e2e-tests.yml

```diff
@@ -6,10 +6,33 @@ on:
   pull_request:
 jobs:
   run-e2e-tests:
-    runs-on: ubuntu-latest
```

Comment:
Hi @jchunkins, 

First off thank you for putting this big effort to improve the OLM CI system by testing across various different components. This PR captures exactly the sort of thing that we would want to run against every OLM and registry PR. 

That being said, I propose the idea of moving this "OLM test infrastructure" into a standalone repo, so that we can make specific github actions that can run these infrastructure tests against any PR in operator-framework. 

Here are some reasons why we don't necessarily want test infrastructure like this in the OLM repo:

- We recently simplified the OLM CI greatly by removing all the openshift specific CI runs which cut down a lot of flakes. Now the OLM CI is quick and easy to run locally as it runs entirely on a local k8s cluster. We think keeping it this way makes the most sense, and want to avoid encoding a lot of steps into our github actions. Tests should be entirely self contained and not require documentation to be able to run. 
- The way this PR changes test cluster provisioning adds new expectations for additional test setup before running go test, which adds more complexity when developing locally. 
- Making a test catalog in this PR is complex. We've struggled ourselves with the problem of building catalogs locally in e2e tests versus hardcoding quay images. I think we want to solve this by using a declarative config test fixture that can then be converted into a catalogsource. But this is still an open problem.

So that being said, what do you think about moving this PR to a standalone repo, importing all the necessary bits, and running the tests this way. From there we can iterate and integrate this into the OLM CI when it stabilizes. I can also check to see if we can make a designated operator-framework repository for this infrastructure to live in once its stabilized. 

More tests is not strictly better, but a few (3-5) of these overarching tests could be valuable and provide more confidence in the OLM suite working as intended.

Reply:
To make this work in a separate repo, we'd need access to opm to create catalogs, but then also OLM builds to run inside a cluster... Maybe getting the [quickstart](https://github.com/operator-framework/operator-lifecycle-manager/tree/master/deploy/upstream/quickstart) files and running it against a minikube/kind cluster? Not sure where the test code comes into play here either. Open to suggestions on how you'd make this work.

Reply:
Because we've already been working on this for many months, the OLM teams initial suggestion that we in fact put it here, I would rather keep with the original design and then tackle a new PR that carves it out into something new and could be enhanced for the new new declarative index design.  If we spend too much more time chasing the ball around, this will never be finished.